core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -119,6 +119,26 @@ private CatalogProperties() {}
       "client.pool.cache.eviction-interval-ms";
   public static final long CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS_DEFAULT =
       TimeUnit.MINUTES.toMillis(5);
+  /**
+   * A comma separated list of elements used, in addition to the hive metastore uri, to compose the
```

Comment:
Actually, I just realized this list of properties is for all Catalog, but we only want this property for HiveCatalog.  There's a lot of other catalogs that can't can use this flag, given how specific it is.  

Do you think we should move this to "CachedClientPool", as it only makes sense there?

I do realize that there's other cached flags in this file currently only used by Hive, but they seem more generic to me and could be re-purposed by other catalog.

Reply:
I agree `CatalogProperties` may not be the best place for the config, but we also have `CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS` here which is also specific to `CachedClientPool`.

I think it's more consistent and intuitive to keep such configs in the same place because they both configures the cache behavior. And I don't see why TTL is more generic than the key when you're using a cache. So I prefer to leave it here for now. What do you think?

Reply:
Yea as I mentioned, I also saw other cached flags currently used by Hive in this file, but I feel they are more generic.  

The reason I felt its more specific than ttl is its javadoc, it looks quite Hive-specific (mention hive metastore uri, ugi, conf).  Maybe we can fix that instead then, though was not sure how, so hence the suggestion to move.  Maybe if you feel strongly, we can put the note "for Hive Catalog, the following keys are supported.."?

Reply:
Hmm, I think key is generic enough for a cache, but what can be used to compose the key is implementation dependent. How about we leave the config in `CatalogProperties` like this:
```java
  /**
   * A comma separated list of elements used, in addition to the {@link #URI}, to compose the
   * key of the client pool cache.
   * 
   * <p>Supported key elements in a Catalog are implementation-dependent.
   */
  public static final String CLIENT_POOL_CACHE_KEYS = "client-pool-cache-keys";
```
Then we move the rest of the javadoc to `CachedClientPool`. Will that be clearer?

Reply:
@szehon-ho PR updated. Please let me know if you have further comments